codegen/src/main/kotlin/software/amazon/smithy/rust/codegen/smithy/RuntimeTypes.kt

```diff
@@ -53,6 +61,14 @@ fun defaultRuntimeCrateVersion(): String {
     }
 }
 
+/**
+ * A mapping for crate name to user specified version.
+ */
+@JvmInline
+value class CrateVersionMap(
```

Comment:
TIL about [`value class`](https://kotlinlang.org/docs/inline-classes.html). Very interesting.

This is the first time we introduce them in the codebase. I wonder why we're not using them? They seem to be better at encoding "transparent" newtypes, like here, something for which we rely on `data class`.

cc someone who actually knows Kotlin @rcoh.

Reply:
no reason not to use them, TIL about them TBH, they seem handy!

We _often_ end up with more than one field which is why `data class` is usually useful but for single-member types this seems great!